src/main/java/seedu/address/ui/panel/calendar/TimelineDayView.java

```diff
@@ -0,0 +1,201 @@
+            if (sameDay(event)) {
+                addEventCard(event);
+            }
+        }
+    }
+
+    /**
+     * Returns true if the event is the same day as the current timeline.
+     * @param event The given event.
+     * @return True if the event is the same day as the current timeline.
+     */
+    private boolean sameDay(EventSource event) {
+        return uiParser.getDay(event.getStartDateTime().getDateTime()).equals(this.day);
+    }
+
+    /**
+     * Adds an event to the timeline.
+     * @param event Represents the given event.
+     */
+    private void addEventCard(EventSource event) {
+        // Gets the event Hour
+        Integer eventHour = uiParser.getHour(event.getStartDateTime().getDateTime());
+        this.eventMap.put(eventHour, event);
+
+        // Creates and add the event card
+        EventCard eventCard = new EventCard(event, uiParser);
+        VBox timeBox = timeBoxes.get(eventHour);
+        timeBox.getChildren().add(eventCard.getRoot());
+
+        // Set Constraints for the grid pane
+        RowConstraints rowConstraints = timelineGrid.getRowConstraints().get(eventHour);
+        rowConstraints.setPrefHeight(timeBox.getHeight() + SPACING);
+        // For the last row
+        if (eventHour.equals(23)) {
+            rowConstraints.setPrefHeight(rowConstraints.getPrefHeight() + SPACING);
+        }
+
+    }
+
+    /**
+     * Resets the timeline by removing all of the eventCards and reset the size.
+     */
+    private void resetTimeline() {
+        eventMap.clear();
+        for (int eventHour = 0; eventHour < 24; eventHour++) {
+            // Reset RowConstraints
+            RowConstraints rowConstraints = timelineGrid.getRowConstraints().get(eventHour);
+            rowConstraints.setPrefHeight(0);
+
+            // Reset EventCards for each hour
+            timeBoxes.get(eventHour).getChildren().clear();
+        }
+    }
+
+    /**
+     * Adds all the timeboxes to the list for easier time reading and writing
+     */
+    private void addTimeBoxes() {
+        this.timeBoxes.add(timeBox0);
+        this.timeBoxes.add(timeBox1);
+        this.timeBoxes.add(timeBox2);
+        this.timeBoxes.add(timeBox3);
+        this.timeBoxes.add(timeBox4);
+        this.timeBoxes.add(timeBox5);
+        this.timeBoxes.add(timeBox6);
+        this.timeBoxes.add(timeBox7);
+        this.timeBoxes.add(timeBox8);
+        this.timeBoxes.add(timeBox9);
+        this.timeBoxes.add(timeBox10);
+        this.timeBoxes.add(timeBox11);
+        this.timeBoxes.add(timeBox12);
+        this.timeBoxes.add(timeBox13);
+        this.timeBoxes.add(timeBox14);
+        this.timeBoxes.add(timeBox15);
+        this.timeBoxes.add(timeBox16);
+        this.timeBoxes.add(timeBox17);
+        this.timeBoxes.add(timeBox18);
+        this.timeBoxes.add(timeBox19);
+        this.timeBoxes.add(timeBox20);
+        this.timeBoxes.add(timeBox21);
+        this.timeBoxes.add(timeBox22);
+        this.timeBoxes.add(timeBox23);
```

Comment:
LOL I see what you mean now. Perhaps it would be better to dynamically append Labels and VBoxes into the GridPane with a for loop. The `TimelineDayView.fxml` file also contains many similar lines because of this.